commands/bootstrap

```diff
@@ -297,6 +332,331 @@ create_self_signed() {
+    LOCAL_CERT_FILE="/etc/mosquitto/ca_certificates/$(basename "$LOCAL_CERT_FILE")"
+    "${EXEC_CMD[@]}" "$TARGET" "$SUDO" chmod 444 "$LOCAL_CERT_FILE"
+    "${EXEC_CMD[@]}" "$TARGET" "$SUDO" chown mosquitto:mosquitto "$LOCAL_CERT_FILE"
+
+    "${EXEC_CMD[@]}" "$TARGET" "$SUDO" mv "$LOCAL_CERT_KEY" /etc/mosquitto/ca_certificates/
+    LOCAL_CERT_KEY="/etc/mosquitto/ca_certificates/$(basename "$LOCAL_CERT_KEY")"
+    "${EXEC_CMD[@]}" "$TARGET" "$SUDO" chmod 400 "$LOCAL_CERT_KEY"
+    "${EXEC_CMD[@]}" "$TARGET" "$SUDO" chown mosquitto:mosquitto "$LOCAL_CERT_KEY"
+
+    ## FIXME: For now nothing is changed on the host itself (so other tools, such as collectd, can connect as before)
+    ##
+    ## "${EXEC_CMD[@]}" "$TARGET" $SUDO tedge config set mqtt.bind.address "127.0.0.1"
+    ## "${EXEC_CMD[@]}" "$TARGET" $SUDO tedge config set mqtt.client.host "127.O.0.1"
+
+    ## FIXME: Mosquitto configuration generated with `mqtt.external` is incorrect
+    ##        see https://github.com/thin-edge/thin-edge.io/issues/2740
+    ##
+    ## "${EXEC_CMD[@]}" "$TARGET" $SUDO tedge config set mqtt.external.bind.address "$TARGET_IP"
+    ## "${EXEC_CMD[@]}" "$TARGET" $SUDO tedge config set mqtt.external.bind.port 1883
+    ## "${EXEC_CMD[@]}" "$TARGET" $SUDO tedge config set mqtt.external.ca_path "$SIGNING_CERT_FILE"
+    ## "${EXEC_CMD[@]}" "$TARGET" $SUDO tedge config set mqtt.external.cert_file "$LOCAL_CERT_FILE"
+    ## "${EXEC_CMD[@]}" "$TARGET" $SUDO tedge config set mqtt.external.key_file "$LOCAL_CERT_KEY"
+
+    ## As a workaround the configuration for external settings is created independently without `tedge`
+
+    cat << EOF >"/tmp/${DEVICE_ID}-tls-listener.conf"
+listener 8883 ${TARGET_IP}
+allow_anonymous false
+require_certificate true
+use_identity_as_username true
+use_username_as_clientid false
+cafile ${SIGNING_CERT_FILE}
+certfile ${LOCAL_CERT_FILE}
+keyfile ${LOCAL_CERT_KEY}
+EOF
+
+    "${SCP_CMD}" "/tmp/${DEVICE_ID}-tls-listener.conf" "$TARGET" "/tmp/tls-listener.conf"
+    "${EXEC_CMD[@]}" "$TARGET" "$SUDO" mv "/tmp/tls-listener.conf" /etc/mosquitto/conf.d
+    "${EXEC_CMD[@]}" "$TARGET" "$SUDO" chown mosquitto:mosquitto /etc/mosquitto/conf.d/tls-listener.conf
+    "${EXEC_CMD[@]}" "$TARGET" "$SUDO" chmod 644 /etc/mosquitto/conf.d/tls-listener.conf
+
+    # apply the new config
+    echo "Restart mosquitto on ${TARGET_DNS}" >&2
+    "${EXEC_CMD[@]}" "$TARGET" $SUDO systemctl restart mosquitto
+}
+
+# configure mqtt on a child device
+configure_mtls_mqtt_client() {
+    TARGET_IP="$1"
+    TARGET_DNS="$2"
+    SIGNING_CERT_FILE="$3"
+    LOCAL_CERT_FILE="$4"
+    LOCAL_CERT_KEY="$5"
+
+    if [ -z "$MAIN_DEVICE" ]; then
+        "${EXEC_CMD[@]}" "$TARGET" $SUDO tedge config set mqtt.device_topic_id "device/main//"
+    else
+        "${EXEC_CMD[@]}" "$TARGET" $SUDO tedge config set mqtt.device_topic_id "device/$DEVICE_ID//"
+    fi
+
+    "${EXEC_CMD[@]}" "$TARGET" $SUDO tedge config set mqtt.client.host "$TARGET_IP"
```

Comment:
Using the DNS entry will make things a little easier to manage especially in a network where devices are using DHCP

Reply:
I tested this PR in a context with no local DNS.